ParseLiveQuery/src/test/java/com/parse/TestParseLiveQueryClient.java

```diff
@@ -413,13 +413,42 @@ public void testSessionTokenSentOnSubscribe() {
 
     @Test
     public void testEmptySessionTokenOnSubscribe() {
-        when(mockUser.getSessionToken()).thenReturn("the token");
         when(webSocketClient.getState()).thenReturn(WebSocketClient.State.CONNECTED);
         parseLiveQueryClient.subscribe(ParseQuery.getQuery("Test"));
         verify(webSocketClient, times(1)).send(contains("\"op\":\"connect\""));
         verify(webSocketClient, times(1)).send(and(
                 contains("\"op\":\"subscribe\""),
-                contains("\"sessionToken\":\"the token\"")));
+                not(contains("\"sessionToken\":"))));
+    }
+
+    @Test
+    public void testClientKeySentOnConnect() throws Exception {
+        Parse.Configuration configuration = new Parse.Configuration.Builder(null)
+                .applicationId("1234")
+                .clientKey("1234")
+                .build();
+        ParsePlugins.reset();
+        ParsePlugins.initialize(null, configuration);
+
+        parseLiveQueryClient = ParseLiveQueryClient.Factory.getClient(new URI(""), new WebSocketClientFactory() {
+            @Override
+            public WebSocketClient createInstance(WebSocketClient.WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+                TestParseLiveQueryClient.this.webSocketClientCallback = webSocketClientCallback;
+                webSocketClient = mock(WebSocketClient.class);
+                return webSocketClient;
+            }
+        }, new ImmediateExecutor());
```

Comment:
Is there any reason why you explicitly change the instance from `@Before`? https://github.com/parse-community/ParseLiveQuery-Android/blob/a285cf83ffdccca9032810fdb06b157e195b4105/ParseLiveQuery/src/test/java/com/parse/TestParseLiveQueryClient.java#L56

The initialization looks same to me.